src/inabox/amp-inabox.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The entry point for AMP Runtime (v0.js) when AMP Runtime = AMP Doc.
+ */
+
+import '../polyfills';
+import {chunk} from '../chunk';
+import {fontStylesheetTimeout} from '../font-stylesheet-timeout';
+import {installPerformanceService} from '../service/performance-impl';
+import {installPullToRefreshBlocker} from '../pull-to-refresh';
+import {installGlobalClickListenerForDoc} from '../document-click';
+import {installStyles, makeBodyVisible} from '../style-installer';
+import {installErrorReporting} from '../error';
+import {installDocService} from '../service/ampdoc-impl';
+import {installCacheServiceWorker} from '../service-worker/install';
+import {stubElements} from '../custom-element';
+import {
+    installAmpdocServices,
+    installBuiltins,
+    installRuntimeServices,
+    adopt,
+} from '../runtime';
+import {cssText} from '../../build/css';
+import {maybeValidate} from '../validator-integration';
+import {maybeTrackImpression} from '../impression';
+import {Inabox} from './inabox';
+
+
+new Inabox(self).init();
+
+// TODO(lannka): only install the necessary services.
+
+/** @type {!./service/ampdoc-impl.AmpDocService} */
+let ampdocService;
```

Comment:
I think we should factor this into individual functions that can be called here to avoid duplicating too much code that often has to be changed together.